src/Plot.tsx

```diff
@@ -1,3 +1,47 @@
-import React from 'react';
+import React, { useReducer } from 'react';
 
-export const Plot = () => <div />;
+import { PlotContext, DispatchContext } from './hooks';
+import type { PlotProps, PlotState, ReducerActions } from './types';
+import { getMax, getMin, splitChildren } from './utils';
+
+function reducer(state: PlotState, action: ReducerActions): PlotState {
+  switch (action.type) {
+    case 'newData': {
+      return {
+        ...state,
+        xMin: getMin(action.value.xMin, state.xMin),
+        xMax: getMax(action.value.xMax, state.xMax),
+        yMin: getMin(action.value.yMin, state.yMin),
+        yMax: getMax(action.value.yMax, state.yMax),
+        labels: [...state.labels, action.value.label],
+      };
```

Comment:
This implementation is problematic because if a series is removed, the boundaries are not updated.
We have to keep the data from all series separated in the state. I think we should generate an id for the serie in the Serie component and use that to identify it in the main state (using a Map or a Record).